routers/routes/routes.go

```diff
@@ -689,14 +689,14 @@ func RegisterRoutes(m *macaron.Macaron) {
 			m.Get(".diff", repo.DownloadPullDiff)
 			m.Get(".patch", repo.DownloadPullPatch)
 			m.Get("/commits", context.RepoRef(), repo.ViewPullCommits)
-			m.Post("/merge", reqRepoPullsWriter, bindIgnErr(auth.MergePullRequestForm{}), repo.MergePullRequest)
-			m.Post("/cleanup", context.RepoRef(), repo.CleanUpPullRequest)
+			m.Post("/merge", context.RepoMustNotBeArchived(), reqRepoPullsWriter, bindIgnErr(auth.MergePullRequestForm{}), repo.MergePullRequest)
+			m.Post("/cleanup", context.RepoMustNotBeArchived(), context.RepoRef(), repo.CleanUpPullRequest)
```

Comment:
In fact, this will delete the branch on the head repository not the base repository. For example. there are some pull requests merged but not clean up on the base repository, but then it's archived. Then should user delete head repository's branch?

Reply:
You mean we should still enable the possiblity to still clean it when its archived?

Reply:
I think yes, need more discuss.